ssl-impl/src/main/java/io/crate/protocols/postgres/ssl/SslCertificateHelper.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.protocols.postgres.ssl;
+
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.logging.Loggers;
+
+import java.security.*;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.*;
+
+public class SslCertificateHelper {
+
+    private static final Logger LOGGER = Loggers.getLogger(SslCertificateHelper.class);
+    private static boolean stripRootFromChain = true; //TODO check
+
+    public static X509Certificate[] exportRootCertificates(final KeyStore ks) throws KeyStoreException {
+        final List<String> aliases = toList(ks.aliases());
+
+        final List<X509Certificate> trustedCerts = new ArrayList<X509Certificate>();
+
+        if(LOGGER.isDebugEnabled()) {
+            LOGGER.debug("No alias given, will trust all of the certificates in the store");
```

Comment:
Yes, particularly because if there are no aliases, then we will return an empty array of trusted certificates; this should lead to trusting _no_ certificate.